### PR TITLE
 Fix qtcreator wizard path and don't assume Qt in SDK

### PR DESCRIPTION
**Pull request by rpannek**
The can_find_template_service_wizard test checked for the wrong path to find the wizard, the first patch fixes the test.

Some of the tests were written so that they assumed that the SDK would have Qt installed in it, but not all SDKs delivered by the PELUX project do have Qt installed, some of them are headless and don't need Qt.

This patch rewrites the test framework in a way so that one can easier add new functionality to the SDE and write new tests for it.

### CODE REVIEW

File: sde-cookbook/qtcreator/configure-qtcreator.py [hunk 1]
```diff
@@ -113,8 +114,7 @@ def add_kit(self):
                         + " --Cxxtoolchain " + self.cxx_toolchain_id
                         + " --Ctoolchain " + self.cc_toolchain_id
                         + " --cmake " + self.cmake_id
-                        + " --qt " + self.qtversion_id
-                        + " --mkspec \"" + os.environ["QMAKESPEC"] + "\""
+                        # + qt_params if use_qt else ""
```
Comment (tobsan) [inline sde-cookbook/qtcreator/configure-qtcreator.py]: Shouldn't this be used?

File: sde-cookbook/sdk/install_sdk.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 SDK_SELF_EXTRACTOR=$1
 PELUX_SDK_DESTINATION="/opt/pelux_sdk"
-PATH_TO_SDK_SELF_EXTRACTOR="${PELUX_SDK_DESTINATION}/${SDK_SELF_EXTRACTOR}"
+PATH_TO_SDK_SELF_EXTRACTOR="${PELUX_SDK_DESTINATION}/`basename ${SDK_SELF_EXTRACTOR}`"
```
Comment (tobsan) [inline sde-cookbook/sdk/install_sdk.sh]: Why this change?

**Review by tobsan — CHANGES_REQUESTED** [review]
Will do a more proper review later, but looks mostly good with some exceptions :)

File: sde-cookbook/qtcreator/configure-qtcreator.py (hunk 1 above)
Comment (rpannek) [reply sde-cookbook/qtcreator/configure-qtcreator.py]: God damn it' you're right, I'll fix!

File: sde-cookbook/sdk/install_sdk.sh (hunk 2 above)
Comment (rpannek) [reply sde-cookbook/sdk/install_sdk.sh]: Ah yeah, perhaps I should add a comment explaining this, because we copy the self-extracting-installer from /vagrant/ to /opt/pelux_sdk and we changed that its not directly in /vagrant/ but instead /vagrant/test/stubs/name_sdk.sh, we only need the name of the installer, not the path.

This was necessary because we moved the stub_sdk.sh from root into test/stubs/full_sdk.sh to not pollute the root with all the stub-sdks. It might not be the most elegant solution but we didn't come up with a better one.

File: sde-cookbook/qtcreator/configure-qtcreator.py (hunk 1 above)
Comment (rpannek) [reply sde-cookbook/qtcreator/configure-qtcreator.py]: More interestingly, why did the tests pass?

Comment (thenor) [reply sde-cookbook/qtcreator/configure-qtcreator.py]: Sounds like dangerous false positives. Have you tried a clean build?

Comment (rpannek) [reply sde-cookbook/qtcreator/configure-qtcreator.py]: I did actually, looking into it manually now.

File: test/test_sdk_installation.py [hunk 3]
```diff
@@ -13,44 +13,78 @@
...
+SCRIPTS_PATH = os.path.dirname(__file__) + "/scripts/"
+
+# Helper functions
+
+def scripts_for(path):
+    return [f for f in os.listdir(SCRIPTS_PATH + path) if f.endswith(".py")]
+
+
+def scripts_path_on_vm(path):
+    return "/vagrant/test/scripts/" + path + "/"
```
Comment (DunderRoffe) [inline test/test_sdk_installation.py]: what path variable means here is ambiguous since it is also in the name. Maybe call it `scripts_dirname` or something instead?

File: test/test_sdk_installation.py [hunk 4]
```diff
@@ -13,44 +13,78 @@
...
         os.system("vagrant destroy -f")
 
 
+@pytest.fixture(scope="class")
+def vagrant_no_qt():
+    try:
+        assert os.system("SDK_FILE_NAME=\"test/stubs/sans_qmake_sdk.sh\" NO_GUI=1 vagrant up") == 0
+        yield
+    finally:
+        os.system("vagrant destroy -f")
```
Comment (DunderRoffe) [inline test/test_sdk_installation.py:68]: Would be nice if the try finally block were extracted to a function that gets called instead? then the what stub_sdk could be a parameter to that function.

Example
```python
def do_vagrant_up(sdk_file_name):
    try:
        assert os.system('SDK_FILE_NAME="test/stubs/{}" NO_GUI=1 vagrant up'.format(sdk_file_name)) == 0
        yield
    finally:
        os.system("vagrant destroy -f")

@pytest.fixture(scope="class")
def vagrant():
    do_vagrant_up("full_sdk.sh")

@pytest.fixture(scope="class")
def vagrant_no_qt():
    do_vagrant_up("sans_qmake_sdk.sh")
```

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: test/test_sdk_installation.py (hunk 4 above)
Comment (rpannek) [reply test/test_sdk_installation.py:68]: I tried this but for some reason, or most probably because I don't quite understand the usage of yield and assert in this, it doesn't work like you proposed, as soon as I move the assert und yield into the other function the tests fail and tell me that I should run `vagrant up`. I think I will leave it like it is for now.

**Comment by rpannek** [comment]
@DunderRoffe would it be ok if I leave it like it is for now?

**Comment by DunderRoffe** [comment]
@rpannek yes it is ok. But could you add an issue after this is merged that it should be fixed?

**Review by DunderRoffe — APPROVED** [review]
